railties/lib/rails/commands/server/server_command.rb

```diff
@@ -10,6 +10,25 @@
 
 module Rails
   class Server < ::Rack::Server
+    module HandlerSuggestion
+      include Rails::Command::Behavior
+
+      # Hard-coding a bunch of handlers here as we don't have a public way of
+      # querying them from the Rack::Handler registry.
+      HANDLERS = %w[cgi fastcgi webrick lsws scgi thin puma unicorn]
+
+      def self.message(handler)
+        suggestions = HANDLERS
```

Comment:
Think this might be clearer as:

```ruby
suggestions = HANDLERS.sort_by { |h| levenshstein_distance(handler, h) }.map(&:inspect)
```

Don't think the `first(2)` makes sense since we use `first` and `second` below — there's only 8 things in the array, we aren't saving anything with it anyway.

Reply:
The only subtle difference with `.map(&:inspect)` is that the message contains double quotes instead of single ones.